StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetFormFactory/PaymentSheetFormFactory.swift

```diff
@@ -665,14 +673,7 @@ extension PaymentSheetFormFactory {
     }
 
     func makeKlarnaCountry(apiPath: String? = nil) -> PaymentMethodElement? {
-        guard let currency = currency else {
-            assertionFailure("Klarna requires a non-nil currency")
-            return nil
-        }
-
-        let countryCodes = Locale.current.sortedByTheirLocalizedNames(
-            KlarnaHelper.availableCountries(currency: currency)
-        )
+        let countryCodes = Locale.current.sortedByTheirLocalizedNames(addressSpecProvider.countries)
```

Comment:
For country and currency mis-match, we will let the server error out at confirm time rather than have hard coded rules in the SDK which often fall out of date.

Reply:
Forgot to mention, I chatted with @fabianlindfors-stripe and aligned on this decision. Afterpay has the same restrictions with no hard coded list client side.